src/utils/analytics.ts

```diff
@@ -15,10 +15,31 @@ type OnChainEvent = {
   txHash: string
 }
 
-export const sendOnChainEvent = async (event: OnChainEvent) => {
+export type EventResponse = {
+  result: string
+  message: string
+}
+
+export const sendOnChainEvent = async (event: OnChainEvent): Promise<EventResponse> => {
+  if (process.env.NODE_ENV !== 'production') {
+    return {
+      result: 'failed',
+      message: 'Not in production environment',
+    }
+  }
+
   try {
     await analyticsInstance.post('/events', { type: 'on-chain', events: [event] })
-  } catch {}
+    return {
+      result: 'success',
+      message: 'Event sent successfully',
+    }
+  } catch {
+    return {
+      result: 'failed',
+      message: 'Failed to send event',
+    }
+  }
```

Comment:
The results seem to be similar in both on-chain and off-chain event. Should we move them to some variable instead?